### DISCUSSION TITLE
New table generates 2 or 3 of everything

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2196] [Help] [answered]
Hi Guys I am having this problem where a new table generated is showing me multiple of the same fields but does not show multiple items on the left panel where I could delete the extra. do you know why this would be happening? if so is there a fix?

![image](https://user-images.githubusercontent.com/80058238/127677187-be376ac7-dfa4-4f0c-8158-6a3eb6645b69.png)

and in deployment the page shows up blank 
![image](https://user-images.githubusercontent.com/80058238/127677482-da7756b1-cac8-4c2d-8b58-a4bf246d3164.png)

**Comment by aptkingston** [comment, marked as answer]
It looks like the builder user is maybe retrieving 2 items with the same ID, and in the deployed app there aren't any results coming back? If you edit the data provider and set the limit to 1 it will fix the duplicate, but it does look like there are maybe 2 items being returned for the same ID.

In the deployed app, if you open dev tools and check the network response I reckon you might see that there's no data being found. Did you click a link in a table to get to that screen?

**Reply by kevmodrome** [reply]
It looks like there's a repeater in there as well?

Edit: never mind, that should be there 🤦‍♂️

**Reply by KernOil** [reply]
@aptkingston ok, great news yes the limit box was null, and changing it to 1 did the trick for that portion. 
![image](https://user-images.githubusercontent.com/80058238/127678949-13c36b2a-2234-44fd-950e-cc850be0f7d7.png)

still having a problem with viewing the page when selecting  view from the table.
this is the network response 
![image](https://user-images.githubusercontent.com/80058238/127679409-5f6c790c-bd8b-4238-880a-1ab5f1e65e8f.png)

**Reply by KernOil** [reply]
I think I should note that this table pulls from a View and not actually a table. Views have not worked great with the SQL connector. I have to put the parent primary key in the view for the view to work but BB does not recognize the key properly and uses it as an input field. I'm not sure if this is where the problem lies.

**Reply by shogunpurple** [reply]
@AndrewJRamay is this the same issue? https://github.com/Budibase/budibase/issues/2161 If so please continue the discussion under that thread. It seems like this is a different issue to the one that you described originally in this post.

**Reply by KernOil** [reply]
however, I can edit the view with no problem in the data tab. so maybe that is not the problem?

**Reply by KernOil** [reply]
Hi, @shogunpurple ide says it's different because before I couldn't edit or submit a field now I can do both. but I guess it is the same basis of the issue.

**Reply by KernOil** [reply]
I also cant view the page anymore in deployment

**Reply by shogunpurple** [reply]
@AndrewJRamay - 

> Views have not worked great with the SQL connector. 

So, it seems like the issue here in general is view support in external data sources. We did not account for this in the original implementation, and the opinionated connector assumes the use of tables. I've updated your other issue to be named: `View support in Plus Data Sources`. Please detail any issues that you are having related to views there, and we can address them as a whole.

I'm going to mark this particular discussion as answered because andrew and kevin have solved your original problem

**Reply by KernOil** [reply]
sounds good Thank you!
